sklearn/metrics/_scorer.py

```diff
@@ -363,6 +377,62 @@ def _factory_args(self):
         return ", needs_threshold=True"
 
 
+class _ContinuousScorer(_BaseScorer):
+    """Scorer taking a continuous response and output a score for each threshold."""
+
+    def __init__(self, score_func, sign, response_method, kwargs):
+        super().__init__(score_func=score_func, sign=sign, kwargs=kwargs)
+        self.response_method = response_method
+
+    def _score(self, method_caller, estimator, X, y_true, sample_weight=None):
+        """Evaluate predicted target values for X relative to y_true.
+
+        Parameters
+        ----------
+        estimator : object
+            Trained estimator to use for scoring.
+
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Test data that will be fed to estimator.predict.
+
+        y_true : array-like of shape (n_samples,)
+            Gold standard target values for X.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        Returns
+        -------
+        score : float
+            Score function applied to prediction of estimator on X.
+        """
+        pos_label = self._get_pos_label()
+        y_score = method_caller(estimator, self.response_method, X, pos_label=pos_label)
+
+        if sample_weight is not None:
+            score_func = partial(self._score_func, sample_weight=sample_weight)
+        else:
+            score_func = self._score_func
+
+        # TODO: this is pretty slow if we have a lot of `potential_thresholds`
```

Comment:
Actually, the overhead here is due to the parameter validation.
Once we remove the nested parameter validation, we would need to have any parallelism.